src/components/ExportDiscourseContext.tsx

```diff
@@ -117,7 +134,10 @@ const getDiscourseContextResultsByDepth = async ({
         currentDepth: currentDepth + 1,
         maxDepth: maxDepth,
         visitedUids,
+        updateProgressWithDelay,
+        progressRefresh,
       });
+      progressRefresh = progressRefresh === 0.9 ? 0.1 : progressRefresh + 0.1;
```

Comment:
Based on this structure I couldn't find a 'meaning' progress segment.
This at least refreshes the UI (to stop the hang) and notifies the user that "something is happening"

Other option would be to rewrite and query by breadth first.